compiler/rustc_mir_transform/src/dest_prop.rs

```diff
@@ -214,6 +214,9 @@ impl<'tcx> MirPass<'tcx> for DestinationPropagation {
             // This is the set of merges we will apply this round. It is a subset of the candidates.
             let mut merges = FxHashMap::default();
 
+            // We put iterated entires into `merges` (another map) and
+            // `merged_locals` (a set), so the insertion order does not matter.
+            #[allow(rustc::potential_query_instability)]
```

Comment:
The insertion order does matter: we only look at the first appartion of a local.